src/main/java/org/folio/circulation/domain/FeeFineRepository.java

```diff
@@ -0,0 +1,50 @@
+package org.folio.circulation.domain;
+
+import static org.folio.circulation.support.http.ResponseMapping.forwardOnFailure;
+import static org.folio.circulation.support.http.ResponseMapping.mapUsingJson;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.client.CqlQuery;
+import org.folio.circulation.support.http.client.PageLimit;
+import org.folio.circulation.support.http.client.Response;
+import org.folio.circulation.support.http.client.ResponseInterpreter;
+
+import io.vertx.core.json.JsonObject;
+
+public class FeeFineRepository {
+  private static final Integer PAGE_LIMIT = 500;
+
+  private final CollectionResourceClient feeFineStorageClient;
+
+  public FeeFineRepository(Clients clients) {
+    feeFineStorageClient = clients.feeFineStorageClient();
+  }
+
+  public CompletableFuture<Result<FeeFine>> getFeeFine(String feeFineOwnerId, String feeFineType) {
+    Result<CqlQuery> typeQuery = CqlQuery.exactMatch("feeFineType", feeFineType);
```

Comment:
Is there a defined set of fee/fine types?

If there is, could this String parameter be replaced with an enumeration (or similar)?

Reply:
Fee/fine types can be either manual or automatic. Manual are created by user. Automatic are pre-defined, but I'm not sure if we have a complete list - overdue fine will be the first one of them. I'm adding this method to find "Overdue fine" fee/fine type (to create it if it doesn't exist), but it also can be used for finding manual fees/fines by name.

Reply:
Is this property only needed by circulation for finding overdue fines?

If so, I think it would be better to make this general method private (or inline it) and have a public method specific to only the type that circulation cares about at the moment.

Either that, or introduce a class/enum representing the type and only define the types that circulation cares about.

Reply:
Done (in the new PR)